test/conformance/device/urDevicePartition.cpp

```diff
@@ -172,3 +172,21 @@ TEST_F(urDevicePartitionTest, PartitionByAffinityDomain) {
         }
     }
 }
+
+TEST_F(urDevicePartitionTest, InvalidNullHandleDevice) {
+    ur_device_partition_property_t props[] = {UR_DEVICE_PARTITION_EQUALLY, 1,
+                                              0};
+    ur_device_handle_t sub_device = nullptr;
+    ASSERT_EQ_RESULT(
+        UR_RESULT_ERROR_INVALID_NULL_HANDLE,
+        urDevicePartition(nullptr, props, 1, &sub_device, nullptr));
+}
+
+TEST_F(urDevicePartitionTest, InvalidNullPointerProperties) {
+    for (auto device : devices) {
+        ur_device_handle_t sub_device = nullptr;
+        ASSERT_EQ_RESULT(
+            UR_RESULT_ERROR_INVALID_NULL_POINTER,
+            urDevicePartition(device, nullptr, 1, &sub_device, nullptr));
```

Comment:
This is an interesting point - In the UR spec we allow the user to partition the devices and return `range(0, NumDevices)`. OpenCL is the opposite:
```CL_INVALID_VALUE if out_devices is not NULL and num_devices is less than the number of sub-devices created by the partition scheme.```
I wonder if we should make it more OpenCL like.

Reply:
I looked at the spec again today, and realized that urDeviceSelectBinary doesn't actually return error as I said. I must have misread it yesterday.

The UR way seems more flexible to me but maybe slightly more confusing. I still find it confusing that 0 is allowed. But maybe it's just a design decision to not have too many checks.

Either way, since this is a place where UR and OpenCL differ, it might be a good idea to write a test that checks for the correct behaviour.

Reply:
Agreed. I've added a test that will request only sub devices in range(0, n_subdevices)